packages/flow-core/src/components/f-document-viewer/f-document-viewer.ts

```diff
@@ -0,0 +1,542 @@
+						<f-div width="hug-content"
+							><f-text
+								variant="para"
+								size="small"
+								.weight=${this.getTextWeight(value.type)}
+								.state=${this.getState(value.type)}
+								id="doc-text"
+								.highlight=${this.searchValue}
+								>${key}</f-text
+							></f-div
+						>
+						<f-div
+							><f-text
+								variant="para"
+								size="small"
+								.state=${this.getState(value.type)}
+								id="doc-text"
+								.highlight=${this.searchValue}
+								>${value.title || value}</f-text
+							></f-div
+						>
+				  </f-div>`}
+		</f-div>`;
+	}
+
+	/**
+	 *
+	 * @param obj complete json object for doc creation
+	 * @returns template for the data given
+	 */
+	traverse(obj: FDocViewerContent) {
+		let htmlTemp = html``;
+		Object.entries(obj).forEach(([key, value]) => {
+			if (typeof value === "object" && value !== null && value.data) {
+				// Recursive call for nested objects
+				const innerHtml = this.traverse(value.data);
+				htmlTemp = html` ${htmlTemp} ${this.getAccordionPanel(key, value, innerHtml)} `;
+			} else {
+				htmlTemp = html` ${htmlTemp} ${this.getTextPanel(key, value as FDocumentStatement)} `;
+			}
+		});
+		return htmlTemp;
+	}
+
+	/**
+	 *
+	 * @param index number or a string which is used to scroll to a partiucular section on scroll event
+	 */
+	scrollToSectionMethod(index: number | string) {
+		const el = this.shadowRoot!.querySelector(`#${this.refName(index)}`) as FAccordion;
+		el.scrollIntoView({
+			behavior: "smooth",
+			block: "start"
+		});
+	}
+
+	/**
+	 * check which section is on view on the screen
+	 */
+	checkInView() {
+		if (this.jumpLinks) {
+			const contentRefs = Object.keys(this.content).map(
+				i => this.shadowRoot?.querySelector(`#${this.refName(i)}`) as FAccordion
+			);
+			const sidebarRefs = Object.keys(this.content).map(
+				i => this.shadowRoot?.querySelector(`#${this.sidebarRefName(i)}`) as FDiv
+			);
+			const handleScroll = () => {
+				let activeSectionIndex = -1;
+				let minDistance = Number.MAX_VALUE;
+
+				for (let i = 0; i < contentRefs.length; i++) {
+					const rect = contentRefs[i].getBoundingClientRect();
+					const distance = Math.abs(rect.top);
+					if (distance < minDistance) {
+						minDistance = distance;
+						activeSectionIndex = i;
+					}
+				}
+
+				sidebarRefs.forEach((sidebar, i) => {
+					if (i === activeSectionIndex) {
+						sidebar.selected = "background";
+						setTimeout(() => {
+							sidebar.scrollIntoView({
+								behavior: "auto",
+								block: "start"
+							});
+						}, 1000);
+					} else {
+						sidebar.selected = "none";
+					}
+				});
+			};
+
+			this.fDocViewerScrollContent?.addEventListener("scroll", handleScroll);
+		}
+	}
+
+	/**
+	 *
+	 * @param obj complete json object for doc creation
+	 * @param level until which level of nesting the data must be displayed
+	 * @returns the filtered data to a particular level
+	 */
+	filterObjectByLevel(obj: FDocViewerContent, level: number) {
```

Comment:
Recursive operations are heavy, can you try adding a `data-level` attribute to either the accordion panel or the text panel?

so that you can hide or show based on the level selected